### ISSUE TITLE
app.py

### ISSUE THREAD

**Issue by Anandh4455** [Anandh4455/Credit-Card-Fraud-Detection-Public#1]
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import classification_report
import streamlit as st

# Step 1: Load and preprocess the data
data = pd.read_csv('credit_card_transactions.csv')

# Handle missing values
data = data.dropna()

# Split the dataset into features (X) and target (y)
X = data.drop('fraudulent', axis=1)
y = data['fraudulent']

# Step 2: Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Step 3: Scale the features
scaler = StandardScaler()
X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)

# Step 4: Train a RandomForest model
model = RandomForestClassifier()
model.fit(X_train, y_train)

# Step 5: Evaluate the model
y_pred = model.predict(X_test)
print(classification_report(y_test, y_pred))

# Step 6: Streamlit UI for prediction
def fraud_detection_ui():
    st.title("Credit Card Fraud Detection")

    # User input fields for new transaction data
    amount = st.number_input("Transaction Amount", min_value=0.0)
    time = st.number_input("Transaction Time", min_value=0)
    # Add other relevant inputs from your dataset, e.g., transaction type, etc.

    if st.button("Check for Fraud"):
        # Process the input data for prediction
        new_data = [[amount, time]]  # Add the rest of the feature data as required
        new_data = scaler.transform(new_data)
        
        # Make the prediction
        prediction = model.predict(new_data)
        
        if prediction == 1:
            st.error("Fraudulent Transaction!")
        else:
            st.success("Valid Transaction.")

# Run the Streamlit app
if __name__ == "__main__":
    fraud_detection_ui()
